src/rgw/rgw_gc.cc

```diff
@@ -128,11 +128,146 @@ int RGWGC::list(int *index, string& marker, uint32_t max, bool expired_only, std
   return 0;
 }
 
-int RGWGC::process(int index, int max_secs)
+class RGWGCIOManager {
+  CephContext *cct;
+  RGWGC *gc;
+
+  struct IO {
+    enum Type {
+      UnknownIO = 0,
+      TailIO = 1,
+      IndexIO = 2,
+    } type{UnknownIO};
+    librados::AioCompletion *c{nullptr};
+    string oid;
+    int index{-1};
+    string tag;
+
+    public:
```

Comment:
the struct is already public here

Reply:
Yes, and this is another style element rather than a technical one: I like to use the access label to group ctors and dtors together away from data and other member functions. Comments like "// constructors here" never feel right to me, and there's no other keyword to indicate it. I'll update it, but that's the reason.